### ISSUE TITLE
Deep watching for computeds?

### ISSUE THREAD

**Issue by jperkelens** [yyx990803/vue#674]
Is there support for deep watching of computed dependencies?

In my case I have a computed whose dependency is an array of objects. In each of these objects (i.e. {selected: true}) is a boolean property, I'd like the computed to return the result of ANDing all the values of this field.

Currently, a change to one of the objects in this array does not trigger a reevaluation of the computed. Is there a way to do this, or is this an unsupported feature?

Edit: Attaching fiddle: http://jsfiddle.net/rmk83v5v/2/

**Comment by jperkelens** [comment]
It works in the example provided, what I've just realized is that since I'm iterating through the list using lodash in my code, it must not register the individual items as dependencies. Since lodash is really convenient, I'd still like a way to get around this.

**Comment by jperkelens** [comment]
I've been able to reproduce this in the fiddle by more closely replicating what I'm doing in my code. I can't tell what exactly the problem is, but it seems that it's something to do with setting of the parent list after compilation.

**Comment by TMiguelT** [comment]
It's not that Vue doesn't support deep watching of dependencies, its that Vue doesn't register `selected` as a dependency because initially it doesn't exist in the list objects. This is exactly the same issue as #660, which I posted a couple of weeks ago.

I've simplified your fiddle to the absolute minimum [here](http://jsfiddle.net/sz7pnvqc/3/), and you'll notice that it still doesn't work (allSelected doesn't change). However if you define the `selected` property at the very start, even if it has a null/undefined value, the code will work, as in [this fiddle](http://jsfiddle.net/3zuthb4p/1/). The only thing I changed here was making list equal to this:
```javascript
list: [
    { selected: undefined },
    { selected: undefined },
    { selected: undefined },
    { selected: undefined },
]
```
